### PR TITLE
Add access token for required layers

### PR DESCRIPTION
**Pull request by JorgeMartinezG**
image

### CODE REVIEW

File: api-flask/app/kobo.py [hunk 1]
```diff
@@ -21,8 +21,23 @@ def get_first(items_list: list):
     return items_list[0] if items_list else None
 
 
+def validate_access_token():
+    """Validate that access token received within the request matches the one in the server."""
+    access_token = request.args.get('accessToken')
+
+    server_access_token = getenv('KOBO_ACCESS_TOKEN')
+
+    if server_access_token is None:
```
Comment (ericboucher) [inline api-flask/app/kobo.py:30]: If the server_access_token is None, should we just accept all requests?

**Comment by wadhwamatic** [comment]
This PR relates to https://github.com/WFP-VAM/prism-frontend/issues/477

**Comment by wadhwamatic** [comment]
@JorgeMartinezG - I need some instructions on how to test this. I made the API locally after adding kobo username and password, but keep getting an error related to the tokens file.

Screen Shot 2022-06-27 at 18 01 58

**Comment by JorgeMartinezG** [comment]
@wadhwamatic Sorry for this.

Once you pull latest changes please apply the following steps:
1. Rebuild docker image `make api`
2. Load the frontend as usual: `yarn start`
3. Select the kobo layer, and take Phnom Penh. For testing, use token **71d8926d-c7c7-4dcf-af92-4374251e3e09**

image

**Comment by wadhwamatic** [comment]
@JorgeMartinezG - I followed the instructions you gave, but still get an error. Screenshots below for frontend and backend

Screen Shot 2022-06-30 at 19 00 51

Screen Shot 2022-06-30 at 19 03 58

**Comment by JorgeMartinezG** [comment]
@wadhwamatic sorry amit, made a mistake. Instead of `make api`, should be `make deploy`

**Comment by wadhwamatic** [comment]
Ok, finally got this working locally @JorgeMartinezG . It looks good when I tested with the example you sent (and for other provinces using the token file info). But if I enter an incorrect password rather than get a message saying its incorrect, I get an error:
Screen Shot 2022-07-01 at 10 42 04

**Comment by wadhwamatic** [comment]
Thanks for the update @JorgeMartinezG. Can we move the invalid credentials message to the authentication modal? Currently, it's not really noticeable since the authentication modal is displayed again, and then once you login correctly, you'll see the 'invalid credentials' message from before. Screenshots below.

Screen Shot 2022-07-05 at 15 59 05

Screen Shot 2022-07-05 at 16 00 05

**Pull request closed without merging**